### PR TITLE
Add missing attributes for the Color Control cluster

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem
Some attributes for the Color Control cluster are missing. Adding a command to chip-tool for the color-control cluster ends up crashing.
All those attributes are marked as optional in Simplicity Studio but there are required in ZAP. It seems reasonable to add them to the tree.

 #### Summary of Changes
* Add the missing attributes for Color Control cluster

### CODE REVIEW

**Comment by vivien-apple** [comment]
FWIW I realised that this PR also needs to update a bunch of examples/ but I would like to make sure the direction is right before copy-pasting those changes everywhere.

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> Adding a command to chip-tool for the color-control cluster ends up crashing.

Wait, why?  Missing attributes should just mean that commands that attempt to touch those attributes are not supported or fail, I would think?  Or did Simplicity Studio let us enable command implementations that assume attributes exist without enabling the corresponding attributes?  Even then, I'd expect attempts to mess with the corresponding attributes to just fail due to them not being found, and error responses to be generated, not crashes.  I'd really like us to look in more detail at what's going on there.

> FWIW I realised that this PR also needs to update a bunch of examples/

It does?  I don't think anything except esp32 wifi-echo implements the color-control cluster...

File: examples/wifi-echo/server/esp32/main/gen/af-gen-event.h [hunk 1]
```diff
@@ -49,8 +48,21 @@
     extern void emberAfPluginIasZoneServerManageQueueEventHandler(void);                                                           \
     extern void emberAfPluginNetworkSteeringFinishSteeringEventHandler(void);                                                      \
     extern void emberAfPluginReportingTickEventHandler(void);                                                                      \
-    extern void emberAfPluginScanDispatchScanEventHandler(void);                                                                   \
     extern void emberAfPluginUpdateTcLinkKeyBeginTcLinkKeyUpdateEventHandler(void);                                                \
+    static void networkEventWrapper(EmberEventControl * control, EmberAfNetworkEventHandler handler, uint8_t networkIndex)         \
```
Comment (bzbarsky-apple) [inline examples/wifi-echo/server/esp32/main/gen/af-gen-event.h:52]: Are these changes due to changes in the silabs Simplicity Studio or SDK versions we're using?  They don't seem relevant to the actual changes being made here....

**Comment by vivien-apple** [comment]
> > Adding a command to chip-tool for the color-control cluster ends up crashing.
> 
> Wait, why? Missing attributes should just mean that commands that attempt to touch those attributes are not supported or fail, I would think? Or did Simplicity Studio let us enable command implementations that assume attributes exist without enabling the corresponding attributes? Even then, I'd expect attempts to mess with the corresponding attributes to just fail due to them not being found, and error responses to be generated, not crashes. I'd really like us to look in more detail at what's going on there.

I believe the required field of Simplicity Studio for the ColorControl cluster are in fact required. If you open the ColorControl cluster in ZAP they appears as required, while they are only optional in Simplicity Studio.

The code crashes because of some asserts. For example: 
https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/color-control-server/color-control-server.c#L167

But it happens for all the missing attributes (basically most the ColorControl cluster commands).

> 
> > FWIW I realised that this PR also needs to update a bunch of examples/
> 
> It does? I don't think anything except esp32 wifi-echo implements the color-control cluster...

I thought the files from gen/ where basically copy-pasted to all the examples. That's better if I'm wrong. It means less work on my side :)

File: examples/wifi-echo/server/esp32/main/gen/af-gen-event.h (hunk 1 above)
Comment (vivien-apple) [reply examples/wifi-echo/server/esp32/main/gen/af-gen-event.h:52]: I believe they are. I'm using a different version of the gecko sdk than the one that is used to generate the initial version. For the record I do used 3.0.0, and the initial version was 2.7.0 . 

I looked into the UI to see if there is something activated that I missed but couldn't find anything relevant.

**Comment by bzbarsky-apple** [comment]
> The code crashes because of some asserts.

Ah, yes, that would do it.  OK, so the basic infrastructure is sane and it's just the color-control cluster code that's asserting instead of actually error-handling, good.

> I thought the files from gen/ where basically copy-pasted to all the examples

They were, as of when we only had an on/off cluster, and that's all those examples want for now.  So yeah, no work needed there.

**Review by bzbarsky-apple — APPROVED** [review]
